segmentation violation while trying to fetch endpoints if annotation service.local=true

We've discovered that our kube-router deployment crashed with the following track trace:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x140291d]

goroutine 118 [running]:
github.com/cloudnativelabs/kube-router/pkg/controllers/routing.(*NetworkRoutingController).nodeHasEndpointsForService(0xc421812a00, 0xc420a2a380, 0x193632a, 0x1c, 0xc42075fdc8)
        /go/src/github.com/cloudnativelabs/kube-router/pkg/controllers/routing/ecmp_vip.go:378 +0x14d
github.com/cloudnativelabs/kube-router/pkg/controllers/routing.(*NetworkRoutingController).getVIPsForService(0xc421812a00, 0xc420a2a380, 0x1, 0x26a1f00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/cloudnativelabs/kube-router/pkg/controllers/routing/ecmp_vip.go:316 +0x107e
github.com/cloudnativelabs/kube-router/pkg/controllers/routing.(*NetworkRoutingController).getVIPs(0xc421812a00, 0x1, 0x19126a1, 0x2, 0x191c62d, 0xb, 0x19126a3, 0x2, 0x1914ede, 0x6)
        /go/src/github.com/cloudnativelabs/kube-router/pkg/controllers/routing/ecmp_vip.go:281 +0x1b4
github.com/cloudnativelabs/kube-router/pkg/controllers/routing.(*NetworkRoutingController).getActiveVIPs(0xc421812a00, 0x0, 0x0, 0x0, 0x0, 0x1a0, 0x0, 0x0, 0x0)
        /go/src/github.com/cloudnativelabs/kube-router/pkg/controllers/routing/ecmp_vip.go:271 +0x34
github.com/cloudnativelabs/kube-router/pkg/controllers/routing.(*NetworkRoutingController).Run(0xc421812a00, 0xc4201b7080, 0xc4200c4180, 0xc42050c7c0)
        /go/src/github.com/cloudnativelabs/kube-router/pkg/controllers/routing/network_routes_controller.go:274 +0x7fc
created by github.com/cloudnativelabs/kube-router/pkg/cmd.(*KubeRouter).Run
        /go/src/github.com/cloudnativelabs/kube-router/pkg/cmd/kube-router.go:156 +0xcc3
```

This is indicating that kube-router crashed while trying to fetch the endpoints on the node, but most likely fail because `address.NodeName` is `nil` [here](https://github.com/cloudnativelabs/kube-router/blob/master/pkg/controllers/routing/ecmp_vip.go#L378).

According to the [type definition of the endpoint addresses](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L4166) anything other than `IP` can be `nil`.

kube-router should handle this gracefully and ignore any endpoints without `NodeName`.

To reproduce, annotate `kube-router.io/service.local=true` on the `kubernetes` service in the `default` namespace, which does not have any `NodeName` for its endpoints:
```
$ kubectl get endpoints -n default kubernetes -o json
{
    "apiVersion": "v1",
    "kind": "Endpoints",
    "metadata": {
        "creationTimestamp": "2019-09-09T13:37:00Z",
        "name": "kubernetes",
        "namespace": "default",
        "resourceVersion": "1",
        "selfLink": "/api/v1/namespaces/default/endpoints/kubernetes",
        "uid": "00000000-aaaa-0000-0000-000000000000"
    },
    "subsets": [
        {
            "addresses": [
                {
                    "ip": "10.0.0.1"
                },
                {
                    "ip": "10.0.0.2"
                },
                {
                    "ip": "10.0.0.3"
                }
            ],
            "ports": [
                {
                    "name": "https",
                    "port": 6443,
                    "protocol": "TCP"
                }
            ]
        }
    ]
}
```

--- a/pkg/controllers/routing/ecmp_vip.go
+++ b/pkg/controllers/routing/ecmp_vip.go
@@ -436,8 +436,14 @@ func (nrc *NetworkRoutingController) nodeHasEndpointsForService(svc *v1core.Serv
 
 	for _, subset := range ep.Subsets {
 		for _, address := range subset.Addresses {
-			if *address.NodeName == nrc.nodeName {
-				return true, nil
+			if address.NodeName != nil {
+				if *address.NodeName == nrc.nodeName {
+					return true, nil
+				}
+			} else {
+				if address.IP == nrc.nodeIP.String() {
+					return true, nil
+				}
 			}
 		}
 	}

